optuna/_deprecated.py

```diff
@@ -101,17 +101,18 @@ def _deprecated_func(func: Callable[[Any], Any]) -> Callable[[Any], Any]:
             # TODO(mamu): Annotate this correctly.
             @functools.wraps(func)
             def new_func(*args: Any, **kwargs: Any) -> Any:
-                warnings.warn(
+                message = (
                     "{0} has been deprecated in v{1}. "
                     "This feature will be removed in v{2}. "
                     "See https://github.com/optuna/optuna/releases/tag/v{1}.".format(
                         name if name is not None else func.__name__,
                         deprecated_version,
                         removed_version,
-                    ),
-                    FutureWarning,
-                    stacklevel=2,
+                    )
                 )
+                if text is not None:
+                    message += " " + text
+                warnings.warn(message, FutureWarning, stacklevel=2)
```

Comment:
Could you add a unit test to confirms that `text` is actually added to the end of the warning message?

The following is just an example:
```python
@pytest.mark.parametrize("text", [None, "", "test", "test" * 100])
def test_deprecation_text_specified(text: Optional[str]) -> None:

    ...
    with pytest.warns(FutureWarning) as record:
        decorated_func()

    assert isinstance(record.list[0].message, Warning)
    if text is not None:
        assert record.list[0].message.args[0].endswith(text)

```